### PR TITLE
Bug/date string conversion

### PR DESCRIPTION
**Pull request by tomtom92**

### CODE REVIEW

File: src/query.js [hunk 1]
```diff
@@ -94,7 +95,12 @@ function createFieldAliases (stats) {
 }
 
 function createStatFeatures (stats) {
+  stats = _.cloneDeep(stats)
+  console.log(stats)
```
Comment (dmfenton) [inline src/query.js]: delete me

File: src/query.js [hunk 2]
```diff
@@ -94,7 +95,12 @@ function createFieldAliases (stats) {
...
   return stats.map(attributes => {
+    for (var key in attributes) {
+      if (key === 'dateField') attributes[key] = new Date(attributes[key]).getTime()
```
Comment (dmfenton) [inline src/query.js]: this is too specific, we need to detect not fields called `dateField` but fields that are either of type: `Date` or are an ISO String

File: src/query.js [hunk 3]
```diff
@@ -115,9 +121,11 @@ function createStatFields (stats) {
 }
 
 function detectType (value) {
+  console.log('value: ', value)
```
Comment (dmfenton) [inline src/query.js]: delete me

File: src/query.js [hunk 4]
```diff
@@ -95,6 +96,11 @@ function createFieldAliases (stats) {
 
 function createStatFeatures (stats) {
   return stats.map(attributes => {
+    for (var key in attributes) {
+      if (attributes[key] instanceof Date || moment(attributes[key], [moment.ISO_8601], true).isValid()) {
```
Comment (tomtom92) [inline src/query.js]: hopefully this is more robust

File: src/query.js (hunk 2 above)
Comment (tomtom92) [inline src/query.js]: check and see if the new line makes more sense

File: src/query.js [hunk 5]
```diff
@@ -95,11 +95,11 @@ function createFieldAliases (stats) {
 }
 
 function createStatFeatures (stats) {
-  stats = _.cloneDeep(stats)
-  console.log(stats)
   return stats.map(attributes => {
     for (var key in attributes) {
```
Comment (dmfenton) [inline src/query.js]: use `const` here

**Pull request merged**